pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -2979,6 +2987,13 @@ def visit_functiondef(self, node: nodes.FunctionDef) -> None:
                 continue
```

Comment:
I'd extract `not is_method()` check from here.

Reply:
Done

Reply:
I meant this check needs to be outside the loop.

```python
if node.is_method():
    for match in _METHOD_MATH:
        ...
else:
    for match in self._function_matchers:
        ...
```

Reply:
Ah makes sense 👍

Reply:
Sorry, I updated the comment again to remove duplicated code.